### PR TITLE
Open Flatmap in Maps interface

### PR DESCRIPTION
**Pull request by alan-wu**
# Description

Opening scaffold and flatmap from file browser and gallery will now be done in the map viewer instead of the scaffold viewer and flatmap viewer respectively.
Here is the link to the ticket - https://www.wrike.com/open.htm?id=970444593

## Type of change

Delete those that don't apply.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This has been deployed [here for testing](https://mapcore-demo.org/current/sparc-app/).

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have utilized components from the Design System Library where applicable
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pages/maps/index.vue [hunk 1]
```diff
@@ -113,18 +207,79 @@ export default {
...
+    },
+    openViewWithQuery: async function() {
+      //Open the map with specific view defined by the query.
+      //First get the DOI information if available
+      if (this.$route.query.dataset_id && this.$route.query.dataset_version) {
+        const url = `${process.env.discover_api_host}/datasets/${this.$route.query.dataset_id}`
+        let datasetUrl = this.$route.query.dataset_version ? `${url}/versions/${this.$route.query.dataset_version}` : url
+        const userToken = this.$cookies.get('user-token')
+        if (userToken) {
+          datasetUrl += `?api_key=${userToken}`
```
Comment (Tehsurfer) [inline pages/maps/index.vue:247]: what is the api key for?

File: pages/maps/index.vue [hunk 2]
```diff
@@ -30,7 +31,87 @@
...
 import PageHero from '@/components/PageHero/PageHero.vue'
+import scicrunch from '@/services/scicrunch'
+
+import { failMessage } from '@/utils/notification-messages'
+
+const getFlatmapEntry = async ( route ) => {
+  const uberonid = route.query.uberonid
+  let organ_name = undefined
+  try {
+    organ_name = await scicrunch.getOrganFromUberonId(uberonid)
```
Comment (Tehsurfer) [inline pages/maps/index.vue:45]: I think something in this is broken, I'll check it out

**Review by Tehsurfer — APPROVED** [review]
Looks good and works well!

I think the panZoomEvents call to the flatmap seems to be running too early so they aren't working for the dynamic markers. But that could be a separate issue.

You can see it by going to this link and opening up the console
https://mapcore-demo.org/current/sparc-app/maps/?type=flatmap&dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig

Overall everything else worked well though and I like the code changes

File: pages/maps/index.vue (hunk 2 above)
Comment (alan-wu) [reply pages/maps/index.vue:45]: Is it from the Scaffold dataset? Scaffold datasets do not have the correct information.

File: pages/maps/index.vue (hunk 1 above)
Comment (alan-wu) [reply pages/maps/index.vue:247]: In case of embargoed dataset.

File: pages/maps/index.vue (hunk 2 above)
Comment (Tehsurfer) [reply pages/maps/index.vue:45]: yeah, I think it is just the way I made that function. I'm looking into it now

**Comment by Tehsurfer** [comment]
I looked into the uberon mapping a little bit more.

It appears to be working locally and on sparc.science

Try this:
https://sparc.science/datasets/flatmapviewer?dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon

vs:
https://mapcore-demo.org/current/sparc-app/maps?type=flatmap&dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon

image

**Comment by alan-wu** [comment]
> I looked into the uberon mapping a little bit more.
> 
> It appears to be working locally and on sparc.science
> 

> Try this: https://sparc.science/datasets/flatmapviewer?dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon
> 
> vs: https://mapcore-demo.org/current/sparc-app/maps?type=flatmap&dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon
> 
> image

It is working but the message is incorrectly displayed. We do not zoom into feature on flatmap, this is intended.

**Comment by alan-wu** [comment]
@Tehsurfer I have fixed the problem with the error message now, please have a look.

https://mapcore-demo.org/current/sparc-app/maps/?type=flatmap&dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon

**Comment by Tehsurfer** [comment]
> > I looked into the uberon mapping a little bit more.
> > It appears to be working locally and on sparc.science
> 
> > Try this: https://sparc.science/datasets/flatmapviewer?dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon
> > vs: https://mapcore-demo.org/current/sparc-app/maps?type=flatmap&dataset_version=3&dataset_id=114&taxo=NCBITaxon%3A9823&uberonid=UBERON%3A0001155&for_species=pig&organ=colon
> > image
> 
> It is working but the message is incorrectly displayed. We do not zoom into feature on flatmap anymore, that is intended. I will fix up the message now.

Oh ok I see! That makes sense. Thank you Alan!

**Pull request merged**